src/modules/price-router/permutations/SequencerPriceRouter.sol

```diff
@@ -0,0 +1,71 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.21;
+
+import { PriceRouter, Registry, ERC20, IChainlinkAggregator } from "src/modules/price-router/PriceRouter.sol";
+
+/**
+ * @title SequencerPriceRouter
+ * @notice Adds sequencer uptime feed safety checks to all PriceRouter pricing calls.
+ * @author crispymangoes
+ */
+contract SequencerPriceRouter is PriceRouter {
+    //============================== ERRORS ===============================
+
+    error SequencerPriceRouter__SequencerDown();
+    error SequencerPriceRouter__GracePeriodNotOver();
+
+    //============================== IMMUTABLES ===============================
+
+    /**
+     * @notice Address for the networks sequencer uptime feed.
+     */
+    IChainlinkAggregator internal immutable sequencerUptimeFeed;
+
+    /**
+     * @notice The amount of time that must pass from when the sequencer comes back online
+     *         to when we can continue pricing again.
+     */
+    uint256 internal immutable gracePeriod;
+
+    constructor(
+        address _sequencerUptimeFeed,
+        uint256 _gracePeriod,
+        address newOwner,
+        Registry _registry,
+        ERC20 _weth
+    ) PriceRouter(newOwner, _registry, _weth) {
+        sequencerUptimeFeed = IChainlinkAggregator(_sequencerUptimeFeed);
+        gracePeriod = _gracePeriod;
+    }
+
+    //============================== Sequencer Uptime Logic ===============================
+
+    /**
+     * @notice Layer 2 chains that use sequencers, can have the sequencer go down. If this happens we do not want
+     *         to price assets, as the datafeeds could be stale, and need to be updated.
+     */
+    function _runPreFlightCheck() internal view override {
+        (
+            ,
+            /*uint80 roundID*/ int256 answer,
+            uint256 startedAt /*uint256 updatedAt*/ /*uint80 answeredInRound*/,
```

Comment:
nit: Are you missing some commas here? It all compiles fine, so this is just semantic I think.

Reply:
Lol I tried fixing it but the linter made it look way worse so I just removed the inline comments

Reply:
That makes sense, we'll resolve it then :)